consensus/safety-rules/src/remote.rs

```diff
@@ -103,3 +107,21 @@ impl<T: Payload> TSafetyRules<T> for RemoteClient<T> {
         lcs::from_bytes(&response)?
     }
 }
+
+pub trait TRemoteClient<T>: Send + Sync {
+    fn request(&mut self, input: SafetyRulesInput<T>) -> Result<Vec<u8>, Error>;
+}
+
+struct LocalService<T> {
+    pub remote_service: Arc<RwLock<RemoteService<T>>>,
+}
+
+impl<T: Payload> TRemoteClient<T> for LocalService<T> {
```

Comment:
Hey, sorry, I'm still confused by the structure. I guess we could either add comments or make the names more self-descriptive. To be clear, it's not about the design (I do understand we need some sort of remote proxy design pattern), it's about the mental load for future readers :).

Reply:
I believe this is fixed in a later commit but perhaps I should reorder them so that a natural flow of review is more ... accessible.